docs/csharp/language-reference/compiler-options/command-line-building-with-csc-exe.md

```diff
@@ -30,93 +30,95 @@ translation.priority.ht:
- [How to: Access Command-Line Arguments Using foreach](../../../csharp/programming-guide/main-and-command-args/how-to-access-command-line-arguments-using-foreach.md)   
- [Main() Return Values](../../../csharp/programming-guide/main-and-command-args/main-return-values.md)
\ No newline at end of file
+>  When you build a project by using the Visual Studio IDE, you can display the **csc** command and its associated compiler options in the **Output** window. To display this information, follow the instructions in [How to: View, Save, and Configure Build Log Files](/visualstudio/ide/how-to-view-save-and-configure-build-log-files) to change the verbosity level of the log data to **Normal** or **Detailed**. After you rebuild your project, search the **Output** window for **csc** to find the invocation of the C# compiler.
+
+ **In this topic**
+
+- [Rules for Command-Line Syntax](#vcconcommand-linebuildinganchor1)
+
+- [Sample Command Lines](#vcconcommand-linebuildinganchor2)
+
+- [Differences Between C# Compiler and C++ Compiler Output](#vcconcommand-linebuildinganchor3)
+
+##  <a name="vcconcommand-linebuildinganchor1"></a> Rules for Command-Line Syntax for the C# Compiler
+The C# compiler uses the following rules when it interprets arguments given on the operating system command line:
+
+- Arguments are delimited by white space, which is either a space or a tab.
+
+- The caret character (^) is not recognized as an escape character or delimiter. The character is handled by the command-line parser in the operating system before it is passed to the argv array in the program.
+
+- A string enclosed in double quotation marks ("string") is interpreted as a single argument, regardless of white space that is contained within. A quoted string can be embedded in an argument.
+
+- A double quotation mark preceded by a backslash (\\") is interpreted as a literal double quotation mark character (").
+
+- Backslashes are interpreted literally, unless they immediately precede a double quotation mark.
+
+- If an even number of backslashes is followed by a double quotation mark, one backslash is put in the argv array for every pair of backslashes, and the double quotation mark is interpreted as a string delimiter.
+
+- If an odd number of backslashes is followed by a double quotation mark, one backslash is put in the argv array for every pair of backslashes, and the double quotation mark is "escaped" by the remaining backslash. This causes a literal double quotation mark (") to be added in argv.
+
+<a name="vcconcommand-linebuildinganchor2"></a>
+## Sample Command Lines for the C# Compiler
+
+- Compiles File.cs producing File.exe:
+
+```
+csc File.cs 
+```
+
+- Compiles File.cs producing File.dll:
+
+```
+csc /target:library File.cs
+```
+
+- Compiles File.cs and creates My.exe:
+
+```
+csc /out:My.exe File.cs
+```
+
+- Compiles all the C# files in the current directory, with optimizations on and defines the DEBUG symbol. The output is File2.exe:
```

Comment:
* Remove comma after "directory"
* "optimizations on" ... Given that it's a preposition it always struck me as a bit weird. We've all see it all over Earth phrased this way. How about instead of "on" using the word "enabled" there?
* \*File2.exe\*